webhook/aws/sns_handler.go

```diff
@@ -0,0 +1,243 @@
+  "Timestamp" : "2012-04-26T20:45:04.751Z",
+  "SignatureVersion" : "1",
+  "Signature" : "EXAMPLEpH+DcEwjAPg8O9mY8dReBSwksfg2S7WKQcikcNKWLQjwu6A4VbeS0QHVCkhRS7fUQvi2egU3N858fiTDN6bkkOxYDVrY0Ad8L10Hs3zH81mtnPk5uvvolIC1CXGu43obcgFxeL3khZl8IKvO61GWB6jI9b5+gLPoBc1Q=",
+  "SigningCertURL" : "https://sns.us-west-2.amazonaws.com/SimpleNotificationService-f3ecfb7224c7233fe7bb5f59f96de52f.pem"
+  }
+
+POST / HTTP/1.1
+x-amz-sns-message-type: Notification
+x-amz-sns-message-id: 22b80b92-fdea-4c2c-8f9d-bdfb0c7bf324
+x-amz-sns-topic-arn: arn:aws:sns:us-west-2:123456789012:MyTopic
+x-amz-sns-subscription-arn: arn:aws:sns:us-west-2:123456789012:MyTopic:c9135db0-26c4-47ec-8998-413945fb5a96
+Content-Length: 773
+Content-Type: text/plain; charset=UTF-8
+Host: example.com
+Connection: Keep-Alive
+User-Agent: Amazon Simple Notification Service Agent
+
+{
+  "Type" : "Notification",
+  "MessageId" : "22b80b92-fdea-4c2c-8f9d-bdfb0c7bf324",
+  "TopicArn" : "arn:aws:sns:us-west-2:123456789012:MyTopic",
+  "Subject" : "My First Message",
+  "Message" : "Hello world!",
+  "Timestamp" : "2012-05-02T00:54:06.655Z",
+  "SignatureVersion" : "1",
+  "Signature" : "EXAMPLEw6JRNwm1LFQL4ICB0bnXrdB8ClRMTQFGBqwLpGbM78tJ4etTwC5zU7O3tS6tGpey3ejedNdOJ+1fkIp9F2/LmNVKb5aFlYq+9rk9ZiPph5YlLmWsDcyC5T+Sy9/umic5S0UQc2PEtgdpVBahwNOdMW4JPwk0kAJJztnc=",
+  "SigningCertURL" : "https://sns.us-west-2.amazonaws.com/SimpleNotificationService-f3ecfb7224c7233fe7bb5f59f96de52f.pem",
+  "UnsubscribeURL" : "https://sns.us-west-2.amazonaws.com/?Action=Unsubscribe&SubscriptionArn=arn:aws:sns:us-west-2:123456789012:MyTopic:c9135db0-26c4-47ec-8998-413945fb5a96"
+  }
+*/
+type MsgAttr struct {
+	Type  string
+	Value string
+}
+type SNSMessage struct {
+	Type             string
+	MessageId        string
+	Token            string
+	TopicArn         string
+	Message          string
+	SubscribeURL     string
+	Timestamp        string
+	SignatureVersion string
+	Signature        string
+	SigningCertURL   string
+	UnsubscribeURL   string
+	MessageAttributes map[string]MsgAttr
+}
+
+// Define handlers for various AWS SNS POST calls
+func (ss *SNSServer) SetSNSRoutes(urlPath string, r *mux.Router, handler http.Handler) {
+	
+	r.HandleFunc(urlPath, ss.SubscribeConfirmHandle).Methods("POST").Headers("x-amz-sns-message-type", "SubscriptionConfirmation")
+	r.Handle(urlPath, handler).Methods("POST").Headers("x-amz-sns-message-type", "Notification")
+	r.HandleFunc(urlPath, ss.UnsubscribeConfirmHandle).Methods("POST").Headers("x-amz-sns-message-type", "UnsubscribeConfirmation")
+
+}
+
+// Subscribe to AWS SNS Topic to receive notifications
+func (ss *SNSServer) Subscribe() bool {
```

Comment:
If other code needs to wait until the subscription is ready, you can use better golang-isms for this.  For example, this function could return a channel or a `sync.WaitGroup` that are used to signal that the subscription is ready.